### PR TITLE
[Merged by Bors] - feat: `positivity` extensions for `Finset.card`, `Fintype.card`

### PR DESCRIPTION
**Pull request by YaelDillies**
Leverage the new `proveFinsetNonempty` prover to write a `positivity` extension about `Finset.card`. Also write the corresponding `positivity` extension for `Fintype.card`.

I put the two new extensions in a new file rather than `Tactic.Positivity.Basic` or `Data.Fintype.Card` because the former doesn't import any finiteness and the latter doesn't import linearly ordered fields. But I don't really mind where the extensions end up.

From LeanAPAP

---

- [x] depends on: #10538

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Positivity/Finset.lean [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  | _ => throwError "not Finset.card"
+
+/-- Extension for `Fintype.card`. `Fintype.card α` is positive if `α` is nonempty. -/
+@[positivity Fintype.card _]
+def evalFintypeCard : PositivityExt where eval {u α} _ _ e := do
+  match u, α, e with
+  | 0, ~q(ℕ), ~q(@Fintype.card $β $instβ) =>
+    let instβno ← synthInstanceQ q(Nonempty $β)
+    assumeInstancesCommute
+    return .positive q(@Fintype.card_pos $β $instβ $instβno)
```
Comment (Vierkantor) [inline Mathlib/Tactic/Positivity/Finset.lean:40]: Should this also have a branch that returns `nonnegative` if proving positivity fails?

File: Mathlib/Tactic/Positivity/Finset.lean [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# Positivity extensions for finsets
+
+This file provides a few `positivity` extensions that cannot be in either the finset files (because
+they don't know about ordered fields) or in `Tactic.Positivity.Basic` (because it doesn't want to
+know about finiteness).
+-/
+
+open Finset
+namespace Mathlib.Meta.Positivity
+open Qq Lean Meta
```
Comment (Vierkantor) [inline Mathlib/Tactic/Positivity/Finset.lean]: Let's visually separate the `namespace` and `open` commands. (I suppose there is no reason to have one `open` inside and another `open` outside the namespace?)
```suggestion
namespace Mathlib.Meta.Positivity

open Qq Lean Meta Finset
```

File: Mathlib/Tactic/Positivity/Finset.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Tactic/Positivity/Finset.lean:40]: No need, the core positivity tactic already proves `0 ≤ n` when `n : Nat`

Comment (Vierkantor) [reply Mathlib/Tactic/Positivity/Finset.lean:40]: Of course, the types are enough!

**Pull request closed without merging**